npe2/manifest/utils.py

```diff
@@ -153,16 +159,131 @@ def __str__(self) -> str:
         return v
 
 
-def import_python_name(python_name: PythonName) -> Any:
+def _import_npe1_shim(shim_name: str) -> Any:
+    """Import npe1 shimmed python_name
+
+    Some objects returned by npe1 hooks (such as locally defined partials or other
+    objects) don't have globally accessible python names. In such cases, we create
+    a "shim" python_name of the form:
+
+    `__npe1shim__.<hook_python_name>_<index>`
+
+    The implication is that the hook should be imported, called, and indexed to return
+    the corresponding item in the hook results.
+
+    Parameters
+    ----------
+    shim_name : str
+        A string in the form `__npe1shim__.<hook_python_name>_<index>`
+
+    Returns
+    -------
+    Any
+        The <index>th object returned from the callable <hook_python_name>.
+
+    Raises
+    ------
+    IndexError
+        If len(<hook_python_name>()) <= <index>
+    """
+
+    assert shim_name.startswith(SHIM_NAME_PREFIX), f"Invalid shim name: {shim_name}"
+    python_name, idx = shim_name[13:].rsplit("_", maxsplit=1)  # TODO, make a function
```

Comment:
I'm not quite sure what the TODO is referring to. I'm not sure I would factor anything out of this function as is.  Should probably avoid the `13` - replace with `len(SHIM_NAME_PREFIX)` etc.